nativelib/src/main/scala/scala/scalanative/regex/Matcher.scala

```diff
@@ -307,17 +307,46 @@ final class Matcher private (private var _pattern: Pattern) {
     }
   }
 
+  def regionSpansInput(): Boolean = {
+    val _regionLength = _regionEnd - _regionStart
+    _regionLength == _inputLength // if true, implies regionStart is 0
+  }
+
+  def anchoredAtStartRegion(): Boolean = (this.start() == regionStart())
+
+  // Design Note:
+  //   Implement region support in a way that it can not break
+  //   prior art. Ugly and paranoid, but contains introduced defects.
+  //
+  //   To aid understanding of the description of "entire input"
+  //   in matches(), lookingAt(), and find() below, mentally substitute
+  //   "entire input region".
+
   // Matches the entire input against the pattern (anchored start and end).
   // If there is a match, {@code matches} sets the match state to describe it.
   //
   // @return true if the entire input matches the pattern
-  def matches(): Boolean = genMatch(0, RE2.ANCHOR_BOTH)
+  def matches(): Boolean = {
```

Comment:
This is the same as `lookingAt` below - perhaps factor out into private method?

Reply:
Thank you for the timely, helpful, and detailed review.  After the second look you prompted,
 I made `matches()` more definitely conform to its semantics.  The code is no longer
identical.

`lookingAt()` is anchored (must match) at the beginning of the region but need not match
the entire region.  `matches()` is defined as the entire region. That is, it is anchored both
at the start and at the end.   

The code in the next submit makes this more evident and also handles a corner case
(reluctant/lazy/shortest_match patterns).

What do you think?

Reply:
I think you are probably happier with it now. 😄